HubSpot.NET/Api/Contact/Dto/ContactHubSpotModel.cs

```diff
@@ -13,186 +14,68 @@
-                { _Phone = Properties.ContainsKey("phone") ? Properties["phone"].Value : string.Empty; }
-                return _Phone;
-            }
-        }
-
-        private string _Address;
+        public string Phone { set; get; }
         [DataMember(Name = "address")]
-        public string Address {
-            set {
-                _Address = value;
-                if (Properties.ContainsKey("address"))
-                    Properties["address"].Value = value;
-            }
-            get
-            {
-                if(string.IsNullOrWhiteSpace(_Address))
-                { _Address = Properties.ContainsKey("address") ? Properties["address"].Value : string.Empty; }
-                return _Address;
-            }
-        }
-
-        private string _City;
+        public string Address { set; get; }
         [DataMember(Name = "city")]
-        public string City {
-            set {
-                _City = value;
-                if (Properties.ContainsKey("city"))
-                    Properties["city"].Value = value;
-            }
-            get
-            {
-                if (string.IsNullOrWhiteSpace(_City))
-                { _City = Properties.ContainsKey("city") ? Properties["city"].Value : string.Empty; }
-                return _City;
-            }
-        }
-
-        private string _State;
+        public string City { set; get; }
         [DataMember(Name = "state")]
-        public string State {
-            set {
-                _State = value;
-                if (Properties.ContainsKey("state"))
-                    Properties["state"].Value = value;
-            }
-            get
-            {
-                if(string.IsNullOrWhiteSpace(_State))
-                { _State = Properties.ContainsKey("state") ? Properties["state"].Value : string.Empty; }
-                return _State;
-            }
-        }
-
-        private string _ZipCode;
+        public string State { set; get; }
         [DataMember(Name = "zip")]
-        public string ZipCode
-        {
-            set
-            {
-                _ZipCode = value;
-                if (Properties.ContainsKey("zip"))
-                    Properties["zip"].Value = value;
-            }
-            get
-            {
-                if (string.IsNullOrWhiteSpace(_ZipCode))
-                { _ZipCode = Properties.ContainsKey("zip") ? Properties["zip"].Value : string.Empty; }
-                return _ZipCode;
-            }
-        }
+        public string ZipCode { set; get; }
 
         [DataMember(Name = "associatedcompanyid")]
         public long? AssociatedCompanyId { get; set; }
 
         [DataMember(Name = "hubspot_owner_id")]
         public long? OwnerId { get; set; }
 
+        // Used for return values
         [DataMember(Name = "properties")]
         public Dictionary<string, ContactProperty> Properties { get; set; } = new Dictionary<string, ContactProperty>();
```

Comment:
I've never understood the reasons for having this on the model (and the associated `LoadProperties` method). Could this not just be constructed at request time?

Reply:
Good question, this is being used for the return values from the `GET` endpoints, because the format of that payload is really different from . It's really for deserialization, I really just made the `LoadProperties` method for someone to be able to manually load the value of the properties into this dictionary if they needed to. I also made it for unit testing. 

The `PropertyValuPairCollection` is instead used for `POST` and `PUT` endpoints. I guess HubSpot thought it was easier for them to use the two formats in different ways.